caching-attribute-service-client/src/main/java/org/hypertrace/core/attribute/service/cachingclient/AttributeCacheContextKey.java

```diff
@@ -0,0 +1,57 @@
+package org.hypertrace.core.attribute.service.cachingclient;
+
+import static java.util.Objects.isNull;
+
+import io.grpc.stub.StreamObserver;
+import io.reactivex.rxjava3.core.Observable;
+import java.util.Objects;
+import java.util.function.Consumer;
+import javax.annotation.Nonnull;
+import org.hypertrace.core.grpcutils.client.GrpcClientRequestContextUtil;
+import org.hypertrace.core.grpcutils.context.RequestContext;
+
+class AttributeCacheContextKey {
+  static AttributeCacheContextKey forCurrentContext() {
+    return forContext(RequestContext.CURRENT.get());
+  }
+
+  static AttributeCacheContextKey forContext(RequestContext context) {
+    assert !isNull(context) : "RequestContext must be set to use Attribute Client";
```

Comment:
assertions might not be enabled by default right? Should we enable them or rather convert this into a guava `Preconditions`?

Reply:
I actually wrote this with requireNonNull and IJ suggested the change. Didn't realize assertions weren't enabled by default, will switch this back and fix that suggestion in my env.